node/config.go

```diff
@@ -2,39 +2,85 @@ package node
 
 import (
 	"io"
+	"os"
 
 	"github.com/BurntSushi/toml"
 
 	"github.com/celestiaorg/celestia-node/node/core"
 	"github.com/celestiaorg/celestia-node/node/p2p"
 )
 
+// ConfigLoader defines a function that loads a config from any source.
+type ConfigLoader func() (*Config, error)
+
 // Config is main configuration structure for a Node.
 // It combines configuration units for all Node subsystems.
 type Config struct {
 	P2P  p2p.Config
 	Core core.Config
 }
 
-// DefaultConfig provides a default Node Config.
-func DefaultConfig() *Config {
+// DefaultConfig provides a default Config for a given Node Type 'tp'.
+func DefaultConfig(tp Type) *Config {
+	switch tp {
+	case Full:
+		return DefaultFullConfig()
+	case Light:
+		return DefaultLightConfig()
+	default:
+		panic("node: unknown Node Type")
+	}
+}
+
+// DefaultFullConfig provides DefaultConfig for Full Node
+func DefaultFullConfig() *Config {
```

Comment:
If I've got the idea correct, we need to expose `DefaultFullConfig` and `DefaultLightConfig` for testing needs. 
Do we plan to expose them(funcs mentioned before) somewhere else? 
is it a good idea to encapsulate them and use `DefaultConfig()` only, if the answer is no, then? 
I would much appreciate to know wdyt about this

Reply:
If you don't see any testing purpose for DefaultFull(Light)Config, I am happy to make those public or even merge in DefaultConfig.

Reply:
🤔 
I thought about using types in testing file itself to exclude the DefaultFullConfig in this config.go
smt like `type DefaultFullConfig = DefaultConfig()` to not have a repetitive scenario in testing. 

But I'm also ok with what you have here.

Reply:
> type DefaultFullConfig = DefaultConfig()

FYI, this won't work, but this will:

> var DefaultFullConfig = DefaultConfig()

Reply:
Yeap. I meant var, sry for typo